lib/src/cln_grpc_base.dart

```diff
@@ -122,6 +122,16 @@ class GRPCClient extends LightningClient {
     return response as T;
   }
 
+  Future<T> listInvoices<T>(
+      {required ListinvoicesRequest params, T Function(Map)? onDecode}) async {
+    /// request to server
+    var response = await stub.listInvoices(params);
+    if (onDecode != null) {
+      return onDecode(toEncode(response.toProto3Json()));
```

Comment:
```suggestion
      return onDecode(response.toProto3Json());
```

You are passing already a JSON

Reply:
can we directly do `return onDecode(response.writeToJsonMap());`?

Reply:
I think you can try, and if the return type is an object we should be able to make a cast to `Map<String, dynamic>`

like `response as Map<String, dynamic>`

I think you can try this and open a new PR to see if the test are happy or now?